drivers/watchdog/wdt_gpio.c

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2018 qianfan Zhao
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <watchdog.h>
+#include <gpio.h>
+
+#define LOG_LEVEL CONFIG_WDT_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(wdt_gpio);
+
+#define WDT_GPIO_PORT			WDT_GPIO_WDT_GPIO0_GPIO_CONTROLLER
+#define WDT_GPIO_PIN			WDT_GPIO_WDT_GPIO0_GPIO_PIN
+#define WDT_HW_MARGIN_MS		WDT_GPIO_WDT_GPIO0_HW_MARGIN_MS
```

Comment:
The marco name such as WDT_GPIO_WDT_GPIO0_GPIO_CONTROLLER are generated by using the node in dts, I don't think this is a good way, but don't know how to deal with this. Hope can get some feedbacks.

```
+	wdt-gpio0 {
+		compatible = "wdt-gpio";
+		label = "wdt-gpio0";
+		gpios = <&gpiob 21 0>;
+		hw-margin-ms = <1600>;
+		status = "ok";
+	};
```